### PR TITLE
Optionally Enforce SLA Remediation Days

### PR DESCRIPTION
**Pull request by blakeaowens**
This PR adds functionality to optionally enforce SLA remediation days for all severities.

 - Add `BooleanField`s to the `SLA_Configuration` model that specify if a given severity will enforce the specified SLA days (`default=True`, to match existing SLA Configuration functionality)
 - Add logic to the finding model's `set_sla_expiration_date` to set the finding `sla_expiration_date` to `None` if its severity is not enforce on the `SLA_Configuration`
 - Update templates displaying SLA configuration information to display `Not Enforced` for any non-enforced SLA days fields

**Addition of new "Enforce (Severity) Finding SLA Days" fields:**
Screenshot 2024-05-10 at 5 04 52 AM

**Viewing all SLAs with Critical Finding SLA Days not enforced:**
Screenshot 2024-05-10 at 5 05 41 AM

**After disabling Critical Finding SLA Days enforcement, critical findings no longer have an SLA expiration date or SLA days remaining:**
Screenshot 2024-05-10 at 5 06 31 AM

**When viewing a Product, severities that do not enforce a given SLA remediation period will show _Not Enforced_:**
Screenshot 2024-05-10 at 5 07 20 AM

[sc-3361]

### CODE REVIEW

**Review by grendel513 — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

File: dojo/api_v2/serializers.py [hunk 1]
```diff
@@ -2938,7 +2938,7 @@ class Meta:
     def validate(self, data):
         async_updating = getattr(self.instance, 'async_updating', None)
         if async_updating:
-            for field in ['critical', 'high', 'medium', 'low']:
+            for field in ['critical', 'enforce_critical', 'high', 'enforce_high', 'medium', 'enforce_medium', 'low', 'enforce_low']:
                 old_days = getattr(self.instance, field, None)
                 new_days = data.get(field, None)
                 if old_days and new_days and (old_days != new_days):
```
Comment (cneill) [inline dojo/api_v2/serializers.py:2944]: If the old value for one of the `enforce_` keys is `True` and the new value is `False` (or the reverse), this first condition will always evaluate to `False` and it wouldn't prevent modification, right?

Comment (blakeaowens) [reply dojo/api_v2/serializers.py]: You are correct. Good catch -- these need to be `if old_days is not None and new_days is not None...`. Resolved in the latest commit.

**Review by cneill — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Comment by coheigea** [comment]
@blakeaowens After updating from 2.34.5 to 2.35.4, I'm seeing this error...it looks like the migration never ran? Do you know how I can handle this error:
```
[19/Jul/2024 12:44:25] ERROR [django.request:241] Internal Server Error: /product/4278
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.UndefinedColumn: column dojo_sla_configuration.enforce_critical does not exist
```

**Pull request merged**